File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/template_extend1_runme.lua
require("import")	-- the import fn
import("template_extend1")	-- import lib into global
te=template_extend1 --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

lb = te.lBaz()
assert(lb:foo() == "lBaz::foo")

db = te.dBaz()
assert(db:foo() == "dBaz::foo")

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/template_extend1_runme.go
package main

import "./template_extend1"

func main() {
	a := template_extend1.NewLBaz()
	b := template_extend1.NewDBaz()

	if a.Foo() != "lBaz::foo" {
		panic(0)
	}

	if b.Foo() != "dBaz::foo" {
		panic(0)
	}
}
